common/OpTestSystem.py

```diff
@@ -1038,14 +1187,80 @@ def sys_ipmi_boot_system_to_petitboot(self, i_console):
         self.cv_IPMI.ipmi_set_boot_to_petitboot()
         self.cv_IPMI.ipmi_power_on()
 
+        self.wait_for_petitboot()
+        self.petitboot_exit_to_shell()
+
+    def wait_for_petitboot(self):
+        console = self.console.get_console()
+        # Wait for petitboot
+        console.expect('Petitboot', timeout=BMC_CONST.PETITBOOT_TIMEOUT)
```

Comment:
Can we catch timeout exception and just raise Boot failure exception, As in running re-provision test, the test expects checking boot behaviour after re-provision completes. The boot is failing(likely a bug) and timeout happens.